crates/tx/src/data/mod.rs

```diff
@@ -183,6 +184,38 @@ pub fn compute_inner_tx_hash(
     Hash(state.finalize_reset().into())
 }
 
+/// Identifier of an inner transaction in a batch.
+pub struct InnerTxId<'tx> {
+    /// Hash of the wrapper transaction, if any.
+    pub wrapper_hash: Option<Cow<'tx, Hash>>,
```

Comment:
A bit unlucky that this is an option, I guess it derives from the fact that `DispatchArgs::Raw` carries an optional wrapper hash?